sourcery_analytics/metrics/utils.py

```diff
@@ -38,12 +38,30 @@ def method_name(method: astroid.nodes.FunctionDef) -> str:
 @nodedispatch
 @validate_node_type(astroid.nodes.FunctionDef)
 def method_lineno(method: astroid.nodes.FunctionDef) -> int:
+    """Returns the line number of the method.
+
+    Not very useful by itself, but can be combined with other metrics for convenience.
```

Comment:
**issue (complexity):** Consider simplifying the implementation for consistency and readability.

While the added functionality is appreciated, I suggest simplifying the implementation to maintain consistency and readability across the codebase. For instance, the extensive docstrings for relatively straightforward functions like `method_lineno` and `method_file` could be made more concise. Additionally, the nested function definition within `method_file` adds unnecessary complexity. Consider integrating the logic directly into the `method_file` function to streamline the flow and improve readability. Simplifying these aspects would make the codebase easier to maintain and understand.